framework/src/problems/EigenProblem.C

```diff
@@ -504,8 +505,13 @@ EigenProblem::solve()
     // could rebuild matrices due to mesh changes or something else.
     _nl_eigen->attachSLEPcCallbacks();
 
-    // Scale eigen vector if necessary
-    preScaleEigenVector();
+    // We need to make sure that ||Bx||_2 is equal to inverse of eigenvalue for
+    // the NEWTON iteration iteration
+    // - On the first solve, we will scale so that this is true for the initial guess
+    // - If the user has a custom normalization, we will also need to re-scale
+    //   back to the required normalization for the NEWTON solve
+    if (_first_solve || _has_normalization)
```

Comment:
This is good if the solution x is not altered outside by users when reentering. But to be safe, I did not have check to make sure we enter the solver properly, i.e. ||Bx||_2 is equal to current eigenvalue. This is debatable. I think it is fine to have this. But we need to add a comment above that `even we do not have this condition, the solver can tolerate it`.